lib/resources/aide_conf.rb

```diff
@@ -0,0 +1,182 @@
+    example "
+    describe aide_conf do
+      its('selection_lines') { should include '/sbin' }
+    end
+
+    describe aide_conf.where { selection_line == '/bin' } do
+      its('rules.flatten') { should include 'r' }
+    end
+
+    describe aide_conf.all_have_rule('sha512') do
+      it { should eq true }
+    end
+    "
+
+    attr_reader :params
+
+    include CommentParser
+
+    def initialize(aide_conf_path = nil)
+      return skip_resource 'The `aide_conf` resource is not supported on your OS.' unless inspec.os.linux?
+      @conf_path = aide_conf_path || '/etc/aide.conf'
+      @content = nil
+      @rules = nil
+      read_content
+    end
+
+    def all_have_rule(rule)
+      # Case when file didn't exist or perms didn't allow an open
+      if @content.instance_of?(String)
+        return false
+      end
+      in_all_lines = true
+      all_lines = parse_conf(@content)
+      all_lines.each do |line|
+        if !line['rules'].include? rule
+          in_all_lines = false
+          break
+        end
+      end
+      in_all_lines
+    end
+
+    filter = FilterTable.create
+    filter.add_accessor(:where)
+          .add_accessor(:entries)
+          .add(:selection_lines, field: 'selection_line')
+          .add(:rules,           field: 'rules')
+
+    filter.connect(self, :params)
+
+    private
+
+    def read_content
+      @content = ''
+      @rules = {}
+      @content = read_file(@conf_path)
+      # Return if @content consists of skip_resource message
+      if @content.instance_of?(String)
+        return @content
+      end
+      @content = filter_comments(@content)
+      @params = parse_conf(@content)
+    end
+
+    def filter_comments(data)
+      content = []
+      data.each do |line|
+        content_line, = parse_comment_line(line, comment_char: '#', standalone_comments: false)
+        content.push(content_line)
+      end
+      content
+    end
+
+    def parse_conf(content)
+      params = []
+      content.each do |line|
+        param = parse_line(line)
+        if !param['selection_line'].nil?
+          params.push(param)
+        end
+      end
+      params
+    end
+
+    def parse_line(line)
+      line_and_rules = {}
+      # Case when line is a rule line
+      if line.include? ' = '
+        parse_rule_line(line)
+      end
+      # Case when line is a selection line
+      if line.start_with?('/', '!', '=')
+        line_and_rules = parse_selection_line(line)
+      end
+      line_and_rules
+    end
+
+    def parse_rule_line(line)
+      line.gsub!(/\s+/, '')
+      rule_line_arr = line.split('=')
+      rules_list = rule_line_arr.last.split('+')
+      rule_name = rule_line_arr.first
+      rules_list.each_index do |i|
+        # Cases where rule respresents one or more other rules
+        if @rules.key?(rules_list[i])
+          rules_list[i] = @rules[rules_list[i]]
+        end
+        check_multi_rule(rules_list, i)
```

Comment:
I'm not a huge fan of passing an object to a method for it to be modified within that method. In this case, `check_multi_rule` actually modifies a value for a key in `rules_list`

Can we modify this to look like this:

```ruby
rules_list[i] = handle_multi_rule(rules_list, i)
```

... and then `handle_multi_rule` returns whatever should be be in that hash value, which, in some cases, may simply be what's already there?